api/repo.go

```diff
@@ -90,87 +91,90 @@ func CreateRepo(c *gin.Context) {
+		r.SetActive(true)
 	}
 
 	if input.GetTimeout() == 0 {
-		input.SetTimeout(constants.BuildTimeoutMin)
+		r.SetTimeout(constants.BuildTimeoutMin)
 	}
 
 	if len(input.GetVisibility()) == 0 {
-		input.SetVisibility(constants.VisibilityPublic)
-	}
-
-	if len(input.GetFullName()) == 0 {
-		input.SetFullName(fmt.Sprintf("%s/%s", input.GetOrg(), input.GetName()))
-	}
-
-	if len(input.GetBranch()) == 0 {
-		input.SetBranch("master")
+		r.SetVisibility(constants.VisibilityPublic)
 	}
 
+	// set default events if no events are passed in
 	if !input.GetAllowPull() && !input.GetAllowPush() &&
 		!input.GetAllowDeploy() && !input.GetAllowTag() &&
 		!input.GetAllowComment() {
-		input.SetAllowPull(true)
-		input.SetAllowPush(true)
+		r.SetAllowPull(true)
+		r.SetAllowPush(true)
 	}
 
 	// create unique id for the repo
 	uid, err := uuid.NewRandom()
 	if err != nil {
-		retErr := fmt.Errorf("unable to create UID for repo %s: %w", input.GetFullName(), err)
+		retErr := fmt.Errorf("unable to create UID for repo %s: %w", r.GetFullName(), err)
 
 		util.HandleError(c, http.StatusServiceUnavailable, retErr)
 
 		return
 	}
 
-	input.SetHash(
+	r.SetHash(
 		base64.StdEncoding.EncodeToString(
 			[]byte(strings.TrimSpace(uid.String())),
 		),
 	)
 
 	// ensure repo is allowed to be activated
-	if !checkAllowlist(input, allowlist) {
-		retErr := fmt.Errorf("unable to activate repo: %s is not on allowlist", input.GetFullName())
+	if !checkAllowlist(r, allowlist) {
+		retErr := fmt.Errorf("unable to activate repo: %s is not on allowlist", r.GetFullName())
 
 		util.HandleError(c, http.StatusForbidden, retErr)
 
 		return
 	}
 
-	// send API call to capture the repo
-	r, err := database.FromContext(c).GetRepo(input.GetOrg(), input.GetName())
-	if err == nil && r.GetActive() {
-		retErr := fmt.Errorf("unable to activate repo: %s is already active", input.GetFullName())
+	// send API call to capture the repo from the database
+	dbRepo, err := database.FromContext(c).GetRepo(r.GetOrg(), r.GetName())
+	if err == nil && dbRepo.GetActive() {
+		retErr := fmt.Errorf("unable to activate repo: %s is already active", r.GetFullName())
 
 		util.HandleError(c, http.StatusConflict, retErr)
 
 		return
 	}
 
 	// check if the repo already has a hash created
-	if len(r.GetHash()) > 0 {
+	if len(dbRepo.GetHash()) > 0 {
 		// overwrite the new repo hash with the existing repo hash
-		input.SetHash(r.GetHash())
+		r.SetHash(dbRepo.GetHash())
 	}
 
 	// send API call to create the webhook
-	url, err := source.FromContext(c).Enable(u, input.GetOrg(), input.GetName(), input.GetHash())
+	_, err = source.FromContext(c).Enable(u, r.GetOrg(), r.GetName(), r.GetHash())
 	if err != nil {
 		retErr := fmt.Errorf("unable to create webhook for %s: %w", r.GetFullName(), err)
 
 		switch err.Error() {
-		case "Repo already enabled":
+		case "repo already enabled":
 			util.HandleError(c, http.StatusConflict, retErr)
 			return
-		case "Repo not found":
+		case "repo not found":
```

Comment:
This is actually a bug fix. `.Enable()` returns the error all lower case (see https://github.com/go-vela/server/blob/v0.6.1/source/github/repo.go#L175-L180). Currently, if there's an error happening at this point the server will return a 5xx error instead of a more accurate 4xx in cases such as enabling a non-existing repo.

Reply:
This is reporting as "Outdated" because the location moved due to changes happening via another merged change. The comment is still valid, however.